### PR TITLE
Arduino Build Support

### PR DESCRIPTION
**Pull request by ksedgwic**
Added Arduino library.properties metadata file.
Also needed some conditional non-compilation of test code which clashes in arduino env.

### CODE REVIEW

File: src/test-utils.c [hunk 1]
```diff
@@ -5,6 +5,11 @@
 //  Licensed under the "BSD-2-Clause Plus Patent License"
 //
 
+// Arduino has two problems with the tests:
+// 1. The test-utils are duplicated w/ other Blockchain Commons packages.
+// 2. The main routine conflicts w/ the actual main.
+#if !defined(ARDUINO)
```
Comment (wolfmcnally) [inline src/test-utils.c]: If you don't want the test code in the Arduino build, isn't there a better way of not including it that doesn't require modifying the source file?

File: src/test-utils.h [hunk 2]
```diff
@@ -8,6 +8,11 @@
 #ifndef TEST_UTILS_H
 #define TEST_UTILS_H
 
+// Arduino has two problems with the tests:
+// 1. The test-utils are duplicated w/ other Blockchain Commons packages.
```
Comment (wolfmcnally) [inline src/test-utils.h]: The test files should *not* be part of the library build. See the makefile.

File: src/test.c [hunk 3]
```diff
@@ -5,6 +5,11 @@
 //  Licensed under the "BSD-2-Clause Plus Patent License"
 //
 
+// Arduino has two problems with the tests:
+// 1. The test-utils are duplicated w/ other Blockchain Commons packages.
```
Comment (wolfmcnally) [inline src/test.c]: See previous comment.

File: src/test.c [hunk 4]
```diff
@@ -5,6 +5,11 @@
...
+// 2. The main routine conflicts w/ the actual main.
+#if !defined(ARDUINO)
```
Comment (wolfmcnally) [inline src/test.c]: See previous comment.

File: src/test-utils.c (hunk 1 above)
Comment (ksedgwic) [reply src/test-utils.c]: In the arduino world everything in the `src` directory gets automatically snarfed up and put in the library.

Could the tests live in an adjacent directory?

File: src/test-utils.h (hunk 2 above)
Comment (ksedgwic) [reply src/test-utils.h]: The arduino world doesn't use config or makefiles.  There is no "build" in the sense that you defined one.  They grab everything in `src`.

For the record, I don't think this makes sense either ...

**Comment by ChristopherA** [comment]
Sorry, I didn’t see the comments. Will discuss tomorrow.

**Pull request merged**